### DISCUSSION TITLE
Can Bioage used to predict the biological age without chronological age in the input?

### DISCUSSION THREAD

**Discussion by qianwang-prenetics** [dayoonkwon/BioAge#15] [Q&A] [answered]
Dear team,

I'm interested in the using of package Bioage to predict the biological age. I have tried to calculate the biological age by function `kdm_calc`, but it returns NA if no chronological age existed in the input data. Attached my code below:
library(BioAge) 
library(dplyr)

> sampledat <- data.frame(sampleID = "2023_1", year = 2023,  gender = 1,  age = NA,  albumin = 4.7, alp = 57, 
>                         creat_umol = 88, glucose_mmol=5.6, lymph = 30, mcv = 90, rdw = 12.8, wbc = 7, crp =0.7)
> biomarkers_selected <- c("albumin", "alp", "creat_umol", "glucose_mmol", 
>                 "lymph", "mcv", "rdw", "wbc", "crp")
> kdm = kdm_nhanes(biomarkers = biomarkers_selected)
> kdm_res <- kdm_calc(data = sampledat, 
>                     biomarkers = biomarkers_selected,
>                     fit = kdm$fit$male,
>                     s_ba2 = kdm$fit$male$s_ba2)

My return is:
> kdm_res$data
  sampleID year gender age albumin alp creat_umol glucose_mmol lymph mcv  rdw wbc crp      kdm kdm_advance
1   2023_1 2023      1  20     4.7  57         88          5.6    30  90 12.8   7 0.7 23.17983    3.179827

**Comment by dayoonkwon** [comment, marked as answer]
No, biological age algorithms rely on chronological age as a fundamental input variable. Without age information, these algorithms cannot calculate biological age because age is a critical factor in modeling the aging process.
